File: backend/src/main/java/com/horain/repository/AgentFeedbackRepository.java
package com.horain.repository;

import com.horain.model.AgentFeedback;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

/**
 * JPA repository for user feedback on agent turns.
 */
public interface AgentFeedbackRepository extends JpaRepository<AgentFeedback, UUID> {

    Optional<AgentFeedback> findByTurnId(UUID turnId);

    List<AgentFeedback> findByRating(String rating);
}

File: backend/src/main/kotlin/com/horain/repository/AgentFeedbackRepository.kt
package com.horain.repository

import com.horain.model.AgentFeedback
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional
import java.util.UUID

/**
 * JPA repository for user feedback on agent turns.
 */
interface AgentFeedbackRepository : JpaRepository<AgentFeedback, UUID> {

    fun findByTurnId(turnId: UUID): Optional<AgentFeedback>

    fun findByRating(rating: String): List<AgentFeedback>
}
